com/course/zenkouv/src/test/java/webDriverTest/Onliner.java

```diff
@@ -0,0 +1,121 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.Wait;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class Onliner {
+
+    private static final String URL = "https://catalog.onliner.by/";
+    private static final String LOGIN = "vitalyzenkou@gmail.com";
+    private static final String PASSWORD = "Qwerty123";
+    private static final String DRIVER = "webdriver.chrome.driver";
+    private static final String DRIVER_PATH = "src\\test\\resources\\chromedriver.exe";
+    private Wait<WebDriver> wait;
+    private WebDriver chromeDriver;
+
+    @BeforeClass
+
+    public void setup() {
+        System.setProperty(DRIVER, DRIVER_PATH);
+        chromeDriver = new ChromeDriver();
+        Duration waitTime = Duration.ofSeconds(40);
+        Duration interval = Duration.ofMillis(5);
+        wait = new FluentWait<>(chromeDriver).withTimeout(waitTime).pollingEvery(interval).ignoring(NoSuchElementException.class);
+    }
+
+    @Test
+    public void loadOnlinerPage() {
+        chromeDriver.get(URL);
+        Assert.assertEquals(chromeDriver.getTitle(), "Каталог Onliner.by");
+    }
+
+    @Test(dependsOnMethods = "loadOnlinerPage")
+    public void authorization() {
+        wait.until(webDriver -> chromeDriver.findElement(By.id(OnlinerLocators.USER_BAR_ID))).click();
+        wait.until(webDriver -> chromeDriver.findElement(By.cssSelector(OnlinerLocators.LOGIN_FIELD_CSS))).sendKeys(LOGIN);
+        chromeDriver.findElement(By.cssSelector(OnlinerLocators.PASSWORD_FIELD_CSS)).sendKeys(PASSWORD);
+        chromeDriver.findElement(By.cssSelector(OnlinerLocators.AUTH_SUBMIT_BUTTON_CSS)).click();
+        Assert.assertTrue(wait.until(webDriver -> chromeDriver.findElements(By.className(OnlinerLocators.PROFILE_CLASS_NAME)).size() > 0),
+                "Authorization failed!");
+    }
+
+    @Test(dependsOnMethods = "authorization")
+    public void addRandomProductToCart() {
+        chooseRandomProductStore();
+        chooseRandomAvailableProduct();
+        addProductToCart();
+        Assert.assertTrue(wait.until(webDriver -> chromeDriver.findElements(By.xpath(OnlinerLocators.CART_XPATH)).size() > 0),
+                "The product wasn't added to cart!");
+    }
+
+    @Test(dependsOnMethods = "addRandomProductToCart")
+    public void checkProductInCart() {
+        wait.until(webDriver -> chromeDriver.findElement(By.xpath(OnlinerLocators.CART_BUTTON_XPATH))).click();
+        Assert.assertTrue(wait.until(webDriver -> chromeDriver.findElements(By.className(OnlinerLocators.CART_PRODUCT_CLASS_NAME)).size() > 0));
```

Comment:
Move this assertion as `IsElementPresent(By By)`  to Validator and reuse in all places